chef: support .NET zero config on linux (#4763)
signalfx/splunk-otel-collector: deployments/chef/CHANGELOG.md

# Changelog

## chef-v0.14.0

- Initial support for [Splunk OpenTelemetry for .NET](https://github.com/signalfx/splunk-otel-dotnet) Auto
  Instrumentation on Linux (x86_64/amd64 only):
  - The .NET SDK is activated by default if the `with_auto_instrumentation` option is set to `true` and
    `auto_instrumentation_version` is `latest` or >= `0.99.0`.

>   - Set the `with_auto_instrumentation_sdks` option to `java` and/or `nodejs` to skip .NET auto instrumentation.

Reviewer: I might suggest we make points like this more abstract now or later as we will keep adding more supported instrumentation instrumentation languages.

Author: Something like?

```suggestion
  - To skip .NET auto instrumentation, configure the `with_auto_instrumentation_sdks` option without `dotnet`.
```
